### PR TITLE
Numerous Balance Changes & Additions

### PR DESCRIPTION
**Pull request by nekomaster1000**
- Cured Jerky's Speed II effect now lasts for 15 seconds.
- Raw Hogchops now poison you with 10 seconds of Infection II, while Cooked Hogchops give you 10 seconds of Slowness II (to balance out their incredible hunger restoration)
- Added the Spirit Eye, which gives you 10 seconds of Glowing when you eat it.
- Changed the Tongue Whip to deal 6 points of damage instead of 3, in addition to making it charge faster by 7 ticks.
- Added the Kinetic Tongue Whip, a Mining Master compatibility weapon. Deals 8 points of damage.
- Removed bizarro 'Dullthorn_block' texture.
- Updated Dullthorns Block texture palette to be consistent with the current Dullthorns textures.
- Updated the Dullthorns Tip texture so that if the tiling breaks and repeats the top texture, it should still look somewhat good.
- Glowsilk now repairs 2000 Durability at the cost of 5 levels, from a previous 200 Durability at 4 levels cost.
- Updated Glowsilk Bow & Ascus Bomb sprites.
- Reduced Blindsights' Luminance bestowing to 5 seconds from 30.
- Blindsights now give you 10 seconds of Jump Boost II on-hit.
- Blindsight Tongue is now consumable for 5 seconds of Jump Boost II.
- Deleted the Skeletal Piglin Loot table :(
- Blackstone Dwarves now only drop an Apple if killed by the Player.
- Blindsights, Shroomloins and Hoglins drop their loot less often.
- Made Shroomloins far harder to anger, removing most blocks from their anger goals.
- It should no longer rain in the Glowstone Canyon (and Delta Shores.)
- Fixed Infection Overlay displaying incorrectly.

To Do:
Kinetic Tongue Whip needs to utilises Knockback III in addition to being given more durability.

Thank you Ludo for helping catch my mistakes and implementing the Hogchop drop-rate change and Kinetic Tongue Whip rendering!

### CODE REVIEW

**Comment by SwanX1** [comment]
Should blindsight attacks give you a positive effect?

File: src/main/java/org/infernalstudios/infernalexp/init/IEItems.java [hunk 1]
```diff
@@ -16,28 +16,18 @@
...
-import org.infernalstudios.infernalexp.items.EntityBucketItem;
-import org.infernalstudios.infernalexp.items.GlowcoalItem;
-import org.infernalstudios.infernalexp.items.GlowsilkBowItem;
-import org.infernalstudios.infernalexp.items.InfernalPaintingItem;
-import org.infernalstudios.infernalexp.items.ItemBase;
-import org.infernalstudios.infernalexp.items.SlurpSoupItem;
-import org.infernalstudios.infernalexp.items.WhipItem;
-import org.infernalstudios.infernalexp.items.IESpawnEggItem;
+import org.infernalstudios.infernalexp.blocks.VerticalSlabBlock;
+import org.infernalstudios.infernalexp.items.*;
```
Comment (SwanX1) [inline src/main/java/org/infernalstudios/infernalexp/init/IEItems.java]: \>:(

**Comment by nekomaster1000** [comment]
The Tongue Stew and Tongue both give you Jump Boost, so it was added to the Blindsight's attack too. You could argue it's somewhat negative when the Blindsight applies it to you as it could momentarily mess with your spacial awareness if you're jumping to fight it.

File: src/main/java/org/infernalstudios/infernalexp/init/IEItems.java [hunk 2]
```diff
@@ -115,7 +121,8 @@
 
     // Tools
     public static final RegistryObject<BowItem> GLOWSILK_BOW = registerItem("glowsilk_bow", () -> new GlowsilkBowItem(new Item.Properties().maxDamage(384).group(InfernalExpansion.TAB)));
-    public static final RegistryObject<WhipItem> BLINDSIGHT_TONGUE_WHIP = registerItem("blindsight_tongue_whip", () -> new WhipItem(IEItemTiers.BLINDSIGHT_TONGUE, 1.0F, -3.4F, new Item.Properties().group(InfernalExpansion.TAB)));
+    public static final RegistryObject<WhipItem> BLINDSIGHT_TONGUE_WHIP = registerItem("blindsight_tongue_whip", () -> new WhipItem(IEItemTiers.BLINDSIGHT_TONGUE, 4.0F, -3.4F, new Item.Properties().group(InfernalExpansion.TAB)));
```
Comment (SwanX1) [inline src/main/java/org/infernalstudios/infernalexp/init/IEItems.java:134]: I thought the damage was balanced, why the change? I mean, you can literally enchant it with sharpness

File: src/main/java/org/infernalstudios/infernalexp/init/IEItems.java [hunk 3]
```diff
@@ -136,4 +143,12 @@ public static void register(IEventBus eventBus) {
...
     }
 
+
+    public static <T extends Item> RegistryObject<T> registerItemConditioned(String name, Supplier<? extends T> itemSupplier, Supplier<? extends T> itemSupplierCompat, String modID) {
+            RegistryObject<T> item = ITEMS.register(name,ModList.get().isLoaded(modID)? itemSupplier:itemSupplierCompat);
+
+            return item;
+    }
+
+
```
Comment (SwanX1) [inline src/main/java/org/infernalstudios/infernalexp/init/IEItems.java]: tooo many empty lines

File: src/main/java/org/infernalstudios/infernalexp/init/IELootModifiers.java [hunk 4]
```diff
@@ -54,6 +54,11 @@ protected HoglinLootModifier(ILootCondition[] conditionsIn) {
...
         @Override
         protected List<ItemStack> doApply(List<ItemStack> generatedLoot, LootContext context) {
+
+            if ((context.getRandom().nextDouble() < 0.6)) {
+                return generatedLoot;
+            }
+
             if (!InfernalExpansionConfig.MobInteractions.USE_HOGCHOPS.getBoolean()) {
                 return generatedLoot;
             }
```
Comment (SwanX1) [inline src/main/java/org/infernalstudios/infernalexp/init/IELootModifiers.java]:
```suggestion
            if (context.getRandom().nextDouble() < 0.6 || !InfernalExpansionConfig.MobInteractions.USE_HOGCHOPS.getBoolean()) {
                return generatedLoot;
            }
```

File: src/main/java/org/infernalstudios/infernalexp/init/IEItems.java (hunk 2 above)
Comment (SwanX1) [inline src/main/java/org/infernalstudios/infernalexp/init/IEItems.java:124]: Why change the attack damage? I think it was balanced before

File: src/main/java/org/infernalstudios/infernalexp/init/IEItems.java (hunk 3 above)
Comment (SwanX1) [inline src/main/java/org/infernalstudios/infernalexp/init/IEItems.java]: Do you really need another method for this?
See my code suggestion above

File: src/main/java/org/infernalstudios/infernalexp/init/IEItems.java [hunk 5]
```diff
@@ -115,7 +121,8 @@
 
     // Tools
     public static final RegistryObject<BowItem> GLOWSILK_BOW = registerItem("glowsilk_bow", () -> new GlowsilkBowItem(new Item.Properties().maxDamage(384).group(InfernalExpansion.TAB)));
-    public static final RegistryObject<WhipItem> BLINDSIGHT_TONGUE_WHIP = registerItem("blindsight_tongue_whip", () -> new WhipItem(IEItemTiers.BLINDSIGHT_TONGUE, 1.0F, -3.4F, new Item.Properties().group(InfernalExpansion.TAB)));
+    public static final RegistryObject<WhipItem> BLINDSIGHT_TONGUE_WHIP = registerItem("blindsight_tongue_whip", () -> new WhipItem(IEItemTiers.BLINDSIGHT_TONGUE, 4.0F, -3.4F, new Item.Properties().group(InfernalExpansion.TAB)));
+    public static final RegistryObject<WhipItem> KINETIC_TONGUE_WHIP = registerItemConditioned("kinetic_tongue_whip", () -> new WhipItem(IEItemTiers.BLINDSIGHT_TONGUE, 6.0F, -3.4F, new Item.Properties().group(InfernalExpansion.TAB)), () -> new WhipItem(IEItemTiers.BLINDSIGHT_TONGUE, 6.0F, -3.4F, new Item.Properties()),"miningmaster");
```
Comment (SwanX1) [inline src/main/java/org/infernalstudios/infernalexp/init/IEItems.java]:
```suggestion
    public static final RegistryObject<WhipItem> KINETIC_TONGUE_WHIP = registerItem("kinetic_tongue_whip", () -> new WhipItem(IEItemTiers.BLINDSIGHT_TONGUE, 6.0F, -3.4F, ModList.get().isLoaded("miningmaster") ? new Item.Properties().group(InfernalExpansion.TAB) : new Item.Properties()));
```

File: src/main/java/org/infernalstudios/infernalexp/init/IEItems.java (hunk 3 above)
Comment (SwanX1) [inline src/main/java/org/infernalstudios/infernalexp/init/IEItems.java]: ~~Skill~~ Formatting issue:
```suggestion
    public static <T extends Item> RegistryObject<T> registerItemConditioned(String name, Supplier<? extends T> itemSupplier, Supplier<? extends T> itemSupplierCompat, String modID) {
        RegistryObject<T> item = ITEMS.register(name, ModList.get().isLoaded(modID) ? itemSupplier : itemSupplierCompat);
        return item;
    }
```

File: src/main/java/org/infernalstudios/infernalexp/init/IELootModifiers.java (hunk 4 above)
Comment (SwanX1) [inline src/main/java/org/infernalstudios/infernalexp/init/IELootModifiers.java]:
```suggestion
            if (context.getRandom().nextDouble() < 0.6 || !InfernalExpansionConfig.MobInteractions.USE_HOGCHOPS.getBoolean()) {
                return generatedLoot;
            }
```

File: src/main/resources/assets/infernalexp/models/item/spirit_eye.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "parent": "item/generated",
   "textures": {
-    "layer0": "infernalexp:item/cured_jerky"
+    "layer0": "infernalexp:item/spirit_eye"
```
Comment (SwanX1) [inline src/main/resources/assets/infernalexp/models/item/spirit_eye.json:4]: lmao what

**Review by SwanX1 — CHANGES_REQUESTED** [review]

**Comment by SwanX1** [comment]
LGTMn't

File: src/main/java/org/infernalstudios/infernalexp/init/IEItems.java (hunk 2 above)
Comment (nekomaster1000) [reply src/main/java/org/infernalstudios/infernalexp/init/IEItems.java:134]: Nobody wanted to use the Whip with it's previous damage output. It's new damage is akin to an Iron Sword, which I feel is appropriate.

**Review by Dariensg — APPROVED** [review]
LGTM

**Pull request merged**